worker/src/main/java/gov/cms/ab2d/worker/processor/ContractProcessorImpl.java

```diff
@@ -87,115 +80,67 @@
-
-                // Periodically check if cancelled
+                // Periodically check if cancelled                if (recordsProcessedCount % cancellationCheckFrequency == 0) {
                 if (recordsProcessedCount % cancellationCheckFrequency == 0) {
-                    isCancelled = hasJobBeenCancelled(progressTracker.getJobUuid());
-                    if (isCancelled) {
-                        log.warn("Job [{}] has been cancelled. Attempting to stop processing the job shortly ... ", progressTracker.getJobUuid());
+                    if (hasJobBeenCancelled(jobUuid)) {
+                        log.warn("Job [{}] has been cancelled. Attempting to stop processing the job shortly ... ", jobUuid);
                         cancelFuturesInQueue(futureHandles);
                         break;
                     }
-
-                    processHandles(futureHandles, progressTracker);
+                    numberOfEobs += processHandles(futureHandles, progressTracker, patients, helper);
                 }
             }
-            awaitTermination(futureHandles, progressTracker);
-
-        } finally {
-            close(helper);
-        }
-
-        handleCancellation(isCancelled);
-
-        // All jobs are done, return the job output records
-        return createJobOutputs(helper.getDataFiles(), helper.getErrorFiles());
-    }
-
-    private StreamHelper createOutputHelper(Path outputDirPath, String contractNumber, FileOutputType outputType, Job job) {
-        try {
-            if (outputType == ZIP) {
-                return new ZipStreamHelperImpl(outputDirPath, contractNumber, getZipRolloverThreshold(),
-                        getRollOverThreshold(), tryLockTimeout, eventLogger, job);
-            } else {
-                return new TextStreamHelperImpl(outputDirPath, contractNumber, getRollOverThreshold(), tryLockTimeout,
-                        eventLogger, job);
-            }
-        } catch (IOException e) {
-            throw new UncheckedIOException(e);
-        }
-    }
-
-    /**
-     * While there are still patient records in progress, sleep for a bit and check progress
-     * @param futureHandles - the running threads
-     * @param progressTracker - the object maintaining the progress of the job
-     */
-    private void awaitTermination(ArrayList<Future<Void>> futureHandles, ProgressTracker progressTracker) {
-        while (!futureHandles.isEmpty()) {
-            sleep();
-            processHandles(futureHandles, progressTracker);
-        }
-    }
-
-    private void close(StreamHelper helper) {
-        if (helper != null) {
-            try {
-                helper.close();
-            } catch (Exception ex) {
-                log.error("Unable to close the helper", ex);
+            while (!futureHandles.isEmpty()) {
+                sleep();
+                numberOfEobs += processHandles(futureHandles, progressTracker, patients, helper);
+                if (hasJobBeenCancelled(jobUuid)) {
```

Comment:
Another artifact of the current architecture, having to poll the job table to see if the API application has performed a cancellation.